src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1654,4 +1662,101 @@ private function doExport($value)
 
         return $export;
     }
+
+    private function generateSignature(\ReflectionFunctionAbstract $r)
+    {
+        $signature = array();
+
+        foreach ($r->getParameters() as $p) {
+            $k = '$'.$p->name;
+            if (method_exists($p, 'isVariadic') && $p->isVariadic()) {
+                $k = '...'.$k;
+            }
+            if ($p->isPassedByReference()) {
+                $k = '&'.$k;
+            }
+            if (method_exists($p, 'getType')) {
+                if ($type = $p->getType()) {
+                    $k = $this->generateTypeHint($type, $r).' '.$k;
+                }
+            } elseif (preg_match('/^(?:[^ ]++ ){4}([a-zA-Z_\x7F-\xFF][^ ]++)/', $p, $type)) {
+                $k = $type[1].' '.$k;
+            }
+            if ($type && $p->allowsNull()) {
+                $k = '?'.$k;
```

Comment:
The `?` is removed when not strictly required, which makes the code compatible with lower PHP versions.